actions/auth.go

```diff
@@ -0,0 +1,104 @@
+package actions
+
+import (
+	"github.com/dgrijalva/jwt-go"
+	"github.com/gobuffalo/buffalo"
+	"github.com/gobuffalo/buffalo/render"
+	"github.com/joepena/monsters/models"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"net/url"
+	"regexp"
+)
+
+type AuthClaims struct {
+	UserId string `json:"user_id"`
+	// Auth payload in here
+	jwt.StandardClaims
+}
+
+var emailRegexp = regexp.MustCompile("^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$")
+
+func getAuthToken(c buffalo.Context) (string, error) {
+	token := jwt.New(jwt.SigningMethodHS256)
+
+	claims := token.Claims.(jwt.MapClaims)
+
+	// Set token claims
+	//claims["exp"] = time.Now().Add(time.Hour * 24).Unix()
+	claims["user_id"] = c.Request().Form.Get("email")
+
+	tokenString, _ := token.SignedString(SERVER_SECRET)
+
+	return tokenString, nil
+}
+
+func createUserHandler(c buffalo.Context) error {
+	f := c.Request().Form
+
+	err := validateCreateUserReqForm(f)
+	if err != nil {
+		return err
+	}
+
+	token, err := getAuthToken(c)
+	if err != nil {
+		log.Error(err)
```

Comment:
Which errors do you want logged?

Reply:
These are general debug errors. I've been using the general rule of thumb of logging all errors so its easy to catch bugs. What do you think of that?